File: Pythonleetcode/Leetcode1331.py
class Solution(object):
    def arrayRankTransform(self, arr):
        ans=[]
        rank_arr=sorted(set(arr))
        ans_dict={}
        rank=1
        for i in rank_arr:
            ans_dict[i]=rank
            rank+=1
        for i in arr:
            ans.append(ans_dict[i])
        return ans


print(Solution().arrayRankTransform(arr = [100,100,100,100]))
print(Solution().arrayRankTransform(arr = [37,12,28,9,100,56,80,5,12]))

File: CppPractice/CppLeetcode/Leetcode1331.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) 
    {
        vector<int> ans;
        
        vector<int> sortArr=arr;
        sort(sortArr.begin(),sortArr.end());
        unordered_map<int,int> theMap;
        int rank=1;
        for(int i=0;i<sortArr.size();i++)
        {
            if(theMap.find(sortArr[i])==theMap.end())
            {
                theMap[sortArr[i]] = rank;
                rank++;
            }
        }
        for(int num:arr)
        {
            ans.push_back(theMap[num]);
        }
        return ans;
        
    }
};